crypto/x509/v3_lib.c

```diff
@@ -274,7 +274,7 @@ int X509V3_add1_i2d(STACK_OF(X509_EXTENSION) **x, int nid, void *value,
         extmp = sk_X509_EXTENSION_value(*x, extidx);
         X509_EXTENSION_free(extmp);
         if (!sk_X509_EXTENSION_set(*x, extidx, ext))
-            return -1;
+            goto m_fail;
```

Comment:
Hmmm, perhaps the X509_EXTENSION_free(extmp); should happen only if the _set() call is successful. Otherwise the old value might stay on the stack and UAF can happen.